src/payment/strategies/square/square-payment-strategy.ts

```diff
@@ -42,14 +70,51 @@ export default class SquarePaymentStrategy extends PaymentStrategy {
             .then(() => super.initialize(options));
     }
 
-    execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
-        const { payment, ...order } = payload;
+    execute(orderRequest: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment } = orderRequest;
 
         if (!payment || !payment.methodId) {
-            throw new InvalidArgumentError('Unable to submit payment because "payload.payment.methodId" argument is not provided.');
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
         }
 
-        const paymentName = payment.methodId;
+        this._paymentMethod = this._store.getState().paymentMethods.getPaymentMethod(payment.methodId);
+
+        if (!this._paymentMethod || !this._paymentMethod.initializationData) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        const { nonce } = this._paymentMethod.initializationData.paymentData;
+
+        const nonceInstrument: NonceInstrument = {
+            nonce,
+        };
+
+        return this._getPaymentData(nonceInstrument)
+            .then(() => {
+                return this._store.dispatch(this._orderActionCreator.submitOrder(orderRequest, options))
+                    .then(() => {
+                        return this._store.dispatch(this._paymentActionCreator.submitPayment({ ...payment, paymentData: nonceInstrument }));
```

Comment:
I think this line will break the normal CC flow for Square, because `initializationData.paymentData.nonce` will only defined for Masterpass flow. What you need to do is to forward the return value of `_getPaymentData` to the payment endpoint, similar to how it was done before. i.e.:

```ts
return this._getPaymentData()
    .then(paymentData =>
        this._store.dispatch(this._orderActionCreator.submitOrder(orderRequest, options)
            .then(() => this._store.dispatch(this._paymentActionCreator.submitPayment({ ...payment, paymentData })))
    );
```